crates/bevy_ecs/src/schedule/executor_parallel.rs

```diff
@@ -125,10 +122,13 @@ impl ParallelSystemExecutor for ParallelExecutor {
 
         ComputeTaskPool::init(TaskPool::default).scope(|scope| {
             self.prepare_systems(scope, systems, world);
+            if 0 == self.should_run.count_ones(..) {
```

Comment:
I don't think this optimization is valid. A system can have a dependency on a system whose run criteria is `no`

Reply:
I was under the impression that if it needs to run, regardless of whether or not it is dependent on another system, it will be marked in should_run. If *any* system needs to run, this check should return false. Is that assumption incorrect?

Reply:
No, you're right. My mental model of what is happening was off. I was mixing how dependencies and should run work.

I think it's possible to add another optimization here then. Where if there's only one system to run we can just run it on the main thread instead of starting the parallel executor.

Reply:
ah wait. nvm that won't work here because we've already spawned the task